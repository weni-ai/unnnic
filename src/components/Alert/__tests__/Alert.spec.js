import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, afterEach, test, vi } from 'vitest';
import UnnnicAlert from '@/components/Alert/Alert.vue';

describe('UnnnicAlert.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(UnnnicAlert, {
      props: {
        version: '2.0',
        title: 'Test Alert',
        text: 'This is an alert message',
        type: 'success',
      },
    });
  });

  afterEach(() => {
    wrapper.unmount();
  });

  test('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('renders the correct component for version 1.1', async () => {
    await wrapper.setProps({ version: '1.1' });
    const versionComponent = wrapper.findComponent({ name: 'Version1dot1' });
    expect(versionComponent.exists()).toBe(true);
  });

  test('does not render alert if version is not 1.0', async () => {
    await wrapper.setProps({ version: '1.2' });
    expect(wrapper.find('.unnnic-alert').exists()).toBe(false);
  });

  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
