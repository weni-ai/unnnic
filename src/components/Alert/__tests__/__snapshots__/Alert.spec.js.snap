// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnnnicAlert.vue > matches the snapshot 1`] = `
"<div data-v-284427ba="" class="unnnic-alert unnnic-alert-position--top-right"><span data-v-26446d8e="" data-v-284427ba="" class="material-symbols-rounded unnnic-icon-scheme--primary unnnic-icon-size--sm" data-testid="material-icon" data-test="unnnic-icon">alert-icon</span>
  <div data-v-284427ba="" class="unnnic-alert__content">
    <div data-v-284427ba="" class="unnnic-alert__title">TEST ALERT</div>
    <div data-v-284427ba="" class="unnnic-alert__text">This is an alert message</div>
  </div>
  <div data-v-284427ba="" class="unnnic-alert__close-text unnnic--clickable">CLOSE</div>
</div>"
`;
